.github/workflows/ci.yml

```diff
@@ -0,0 +1,173 @@
+name: CI
+
+on: push
+
+env:
+  # Would be more usual to set R_LIBS_USER, but R uses R_LIBS first if present
+  # ...and it's always present here, because the rocker/tidyverse base image
+  # checks at R startup time for R_LIBS and R_LIBS_USER, sets both if not found
+  R_LIBS: ~/R/library
+  _R_CHECK_LENGTH_1_CONDITION_: true
+  _R_CHECK_LENGTH_1_LOGIC2_: true
+  # Avoid compilation check warnings that come from the system Makevars, not us
+  # See https://stat.ethz.ch/pipermail/r-package-devel/2019q2/003898.html
+  _R_CHECK_COMPILATION_FLAGS_KNOWN_: -Wformat -Werror=format-security -Wdate-time
+
+jobs:
+
+  build:
+    runs-on: ubuntu-latest
+    container: pecan/depends:develop
+    steps:
+    - name: check git version
+      id: gitversion
+      run: |
+        v=$(git --version | grep -oE '[0-9\.]+')
+        v='cat(numeric_version("'${v}'") < "2.18")'
+        echo "##[set-output name=isold;]$(Rscript -e "${v}")"
+    - name: upgrade git if needed
+      # Hack: actions/checkout wants git >= 2.18, rocker 3.5 images have 2.11
+      # Assuming debian stretch because newer images have git >= 2.20 already
+      if: steps.gitversion.outputs.isold == 'TRUE'
+      run: |
+        echo 'deb http://deb.debian.org/debian stretch-backports main' >> /etc/apt/sources.list
+        apt-get update
+        apt-get -t stretch-backports upgrade -y git
```

Comment:
Sure seems like there ought to be a simpler approach: If git is old, `actions/checkout` falls back to retrieving the source as a tarball instead of cloning the repo, so really all this work is "just" to ensure the `tree-is-clean` action below finds a Git repository to work on.

Reply:
Probably one better option will be to have the latest version of git set up inside all the base images.

Reply:
@rahul799 that makes sense to me. @robkooper do you see any issues with updating Git on at `depends` build time, potentially from backports in the case of images built from Stretch?

Reply:
Moved this to its own issue to handle later.